### PR TITLE
cli: update backend plugin template

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

This updates the backend plugin template in the CLI, both to remove old patterns such as exporting `createRouter`, but also to provide a lot more guidance for building your first backend plugin.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/cli/templates/default-backend-plugin/dev/index.ts.hbs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { createBackend } from '@backstage/backend-defaults';
+import { mockServices } from '@backstage/backend-test-utils';
+import { catalogServiceMock } from '@backstage/plugin-catalog-node/testUtils';
+
+// TEMPLATE NOTE:
+// This is the development setup for your plugin that wires up a
+// minimal backend that can use both real and mocked plugins and services.
+//
+// Start up the backend by running `yarn start` in the package directory.
+// It's it's up and running, try out the following requests:
```
Comment (freben) [inline packages/cli/templates/default-backend-plugin/dev/index.ts.hbs:10]:
```suggestion
// Once it's up and running, try out the following requests:
```

File: packages/cli/templates/default-backend-plugin/src/plugin.test.ts.hbs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  startTestBackend,
+} from '@backstage/backend-test-utils';
+import { {{pluginVar}} } from './plugin';
+import request from 'supertest';
+import { catalogServiceMock } from '@backstage/plugin-catalog-node/testUtils';
+
+// TEMPLATE NOTE:
+// Plugin tests are integration tests for your plugin, ensuring that all pieces
+// work together end-to-end. You can still mock injected backend services
+// however, just like anyway that installs your plugin might replace the
```
Comment (freben) [inline packages/cli/templates/default-backend-plugin/src/plugin.test.ts.hbs:12]:
```suggestion
// however, just like anyone who installs your plugin might replace the
```

Is this what you meant?

File: packages/cli/templates/default-backend-plugin/tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "@backstage/cli/config/tsconfig.json",
   "include": [
-    "src",
+    "src2",
```
Comment (freben) [inline packages/cli/templates/default-backend-plugin/tsconfig.json:4]: Whaa

**Review by freben — COMMENTED** [review]
Awesome that you're doing this! Just drive-by-noted some things, will check closer later

File: packages/cli/templates/default-backend-plugin/tsconfig.json (hunk 3 above)
Comment (Rugvip) [reply packages/cli/templates/default-backend-plugin/tsconfig.json:4]: oh wow, that's VSCode updating that under the hood as I was renaming to keep the old template as a reference

**Review by freben — APPROVED** [review]
Nice! Small optionals

File: packages/cli/templates/default-backend-plugin/src/plugin.ts.hbs [hunk 4]
```diff
@@ -2,7 +2,9 @@ import {
   coreServices,
   createBackendPlugin,
 } from '@backstage/backend-plugin-api';
-import { createRouter } from './service/router';
+import { createRouter } from './router';
+import { catalogServiceRef } from '@backstage/plugin-catalog-node/alpha';
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/cli/templates/default-backend-plugin/src/plugin.ts.hbs:6]: We don't export in from the main entry, do we?

**Comment by Rugvip** [comment]
Gonna merge an follow up just to avoid conflicts with #27104

File: packages/cli/templates/default-backend-plugin/src/plugin.ts.hbs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/cli/templates/default-backend-plugin/src/plugin.ts.hbs:6]: As discussed, not gonna stress this particular bit right now, but we ofc want to move this to stable pretty quick too

**Pull request merged**